### ISSUE TITLE
FTB Revelation Freezes During Construction of Forge Mod Loader

### ISSUE THREAD

**Issue by Lukefnz** [FTBTeam/FTB-Revelation-2#249]
* **Modpack Version**: FTBRevelation-3.0.1-1.12.2.zip - (release) - (current)

* **Issue**: Game Freezes when it reaches; Loading - Constructing Mods 1/7, Construction - Forge Mod Loader 3/229

* **Link to Log or Crash File Paste**: https://paste.feed-the-beast.com/view/72850098

* **Is it Repeatable?**: Repeatable every Time

* **Mod/s Affected**: N/A?

* **Known Fix**: N/A

Process:
1) Downloaded and installed Twitch App for Windows
2) Logged in and clicked on 'Mods'
3) Let Twitch install Minecraft
4) Installed FTB Revelations (see above for version)
5) Went into Twitch Settings > Minecraft, and changed Available Memory to 7.168GB
6) Went into FTB Revelations Settings > Profile Options and changed changed from Use System Memory Settings to 7.177GB (closest I could get it)
7) Refreshed Twitch App and confirmed above setting changes remained
8) Exited Twitch App and re-opened and confirmed setting changes remained
9) Clicked play (still in twitch app)
10) Clicked play (now in Minecraft Launcher)
11) Game Freezes when it reaches; Loading - Constructing Mods 1/7, Construction - Forge Mod Loader 3/229
12) Attempted Reboot of Minecraft, Twitch, Computer - to no effect
13) Attempted uninstalling and reinstalling Modpack and Twitch - to no effect
14) After checking forums (including Issue 231 here) I verified I had latest Java, but Twitch does not seem to allow me to change from Java 1.0.8.51 (64 bit) to Java 1.0.8.211
15) At a loss, making this post.

**Comment by rune-san** [comment]
This here:
`[18:40:42] [Client thread/INFO]: Attempting connection with missing mods [minecraft, mcp, FML, forge, openmodscore, foamfixcore, randompatches, actuallyadditions, actuallybaubles, advanced_machines, advanced_solar_panels, ae2wtlib, akashictome, appleskin, appliedenergistics2, architecturecraft, aroma1997core, aroma1997sdimension, astralsorcery, autoreglib, badwithernocookiereloaded, base, baubles, bdlib, betteradvancements, betterbuilderswands, bibliocraft, binniecore, botany, binniedesign, extrabees, extratrees, genetics, biomesoplenty, blockcraftery, bloodmagic, bookshelf, botania, brandonscore, buildcraftcompat, buildcraftbuilders, buildcraftcore, buildcraftenergy, buildcraftfactory, buildcraftlib, buildcraftrobotics, buildcraftsilicon, buildcrafttransport, careerbees, cctweaked, computercraft, chameleon, chesttransporter, chisel, chiselsandbits, clumps, codechickenlib, cofhcore, cofhworld, compactsolars, conarm, controlling, cookingforblockheads, ctgui, crafttweaker, crafttweakerjei, ctm, custommainmenu, danknull, darkutils, deepresonance, defaultoptions, diethopper, dirt2path, draconicevolution, eleccoreloader, eleccore, endercore, enderio, enderiobase, enderioconduitsappliedenergistics, enderioconduitsopencomputers, enderioconduitsrefinedstorage, enderioconduits, enderiointegrationforestry, enderiointegrationtic, enderiointegrationticlate, enderiomachines, enderiopowertools, enderstorage, engineersdoors, engineersworkshop, environmentaltech, etlunar, exchangers, extrautils2, bigreactors, fastbench, fencejumper, flatcoloredblocks, fluxnetworks, foamfix, forestry, forgelin, microblockcbe, forgemultipartcbe, minecraftmultipartcbe, ftbguides, ftblib, ftbutilities, ftbbackups, gendustry, gravisuite, guideapi, harvest, immersivecables, immersivepetroleum, ic2, industrialforegoing, initialinventory, integrationforegoing, inventorytweaks, ironbackpacks, ironchest, jehc, jei, jeibees, jeiintegration, journeymap, jeresources, letsencryptcraft, longfallboots, lostcities, magicbees, mantle, mcjtylib_ng, mcmultipart, minetogether, minetogetherserver, modnametooltip, modtweaker, moreoverlays, morphtool, morpheus, mousetweaks, mrtjpcore, mtlib, mysticallib, natura, naturescompass, nice, neid, notenoughwands, openblocks, openmods, p455w0rdslib, harvestcraft, placebo, platforms, projectred-core, projectred-fabrication, projectred-integration, projectred-transmission, projectred-illumination, projectred-expansion, projectred-relocation, projectred-transportation, projectred-exploration, quark, quickleafdecay, railcraft, rangedpumps, reborncore, rebornstorage, redstonearsenal, redstoneflux, refinedstorage, refinedstorageaddons, xreliquary, resourceloader, rftools, rftoolscontrol, rftoolspower, rustic, shadowmc, simplevoidworld, simplyjetpacks, snad, sonarcore, soulshardsrespawn, stevescarts, storagedrawers, storagedrawersextra, tconstruct, teslacorelib, teslacorelib_registries, thaumcraft, tcinventoryscan, thaumicjei, theoneprobe, thermalcultivation, thermaldynamics, thermalexpansion, thermalfoundation, thermalinnovation, thutcore_compat, thutcore, thuttech, tinkertoolleveling, tipthescales, topaddons, torchmaster, translocators, twilightforest, unidict, universalmodifiers, valkyrielib, wanionlib, waystones, wct, woot, xnet, xtones, yabba, zerocore, immersiveengineering, industrialwires, reauth, shetiphiancore] at CLIENT`

Indicates you're missing the mods. You've either got a bad install or you might be launching the wrong Minecraft. Sometimes, if using the native launcher, Minecraft will switch from Minecraft Forge to just native Minecraft. If that happens, you'll see a lack of Minecraft Forge on the green button you click to launch.

For using the latest Java, there's a Google Doc showing the process to add an argument to Twitch's Java settings to have it use the latest version: https://docs.google.com/document/d/e/2PACX-1vR5muyi1m7Zy_qSwrzeb9e3tbbRlksguNUb_Pvm3pn9eDpsIkUj5qQUq6WFPPdEoVWZq91WvvH57KzC/pub

**Comment by Lukefnz** [comment]
Thank you for your help.
I am a fan of explicit process lists so forgive me but here's another one.

1) Re-ran FTB Revelations to check to see if there is "a lack of Minecraft Forge on the green button you click to launch." See image below for result. It appears to me to be fine.
![minecraft 1](https://user-images.githubusercontent.com/51472321/59139701-efcb3200-89e8-11e9-8a7c-76d465952c31.PNG)
2) Clicked "Repair Profile" (didn't really expect this to work but wth), re-ran game, same freeze as usual.
3) Deleted the Modpack (and backups) and re-installed the release version (3.0.1). Re-ran game, got the same freeze and the log appears to be identical (Attempting connection with missing mods).
4) Followed the steps provided to force twitch to use my up-to-date version of Java, re-ran game to see if that changed anything, game still froze at the same place and the log appears to be the same again (Attempting connection with missing mods).

My thoughts are that there may be some issue with the installation but the lack of other threads on this issue suggests that this is unlikely. Perhaps there's an issue with the native launcher, but the error you describe that would result from the native launcher does not seem to be occurring.

What do you think?

I see I can change the Launcher to JAR. Perhaps that would help? I'm hesitant to do so since it is a work- around rather than a fix. There may not be a fix so if suggested then I can make that change.

**Comment by rune-san** [comment]
It looks like for the client, the initial connection with missing mods is expected, so it scraps what I said earlier. I just tried it on a virtual machine in my own environment. That being said, obviously it's not normal, nor specifically an FTB issue. Have you re-attempted to set your Java memory allocation to something more usable? Some applications have issues if the allocations aren't a power of 2. A more reasonable allocation of 6144MB may yield better results. If you have all your instances using the same Memory allocation, then you only need to set it in the Twitch launcher settings, not on your profile settings.

**Comment by Lukefnz** [comment]
Still freezing same place.

1) I set Twitch to 6144MB and modpack profile to system memory settings. Same old Freeze. Although I noticed where the loading screen shows how memory is being used the number was less than 6144MB (it was 5.~7MB).
 2) I set Twitch to 6912MB in the hopes that the reduced minecraft loading screen would now read 6144MB, re-ran game, minecraft loader did in fact read 6144MB (6912MB was a lucky guess) but the game froze again, same place.
